src/com/owncloud/android/lib/resources/files/ChunkedUploadRemoteFileOperation.java

```diff
@@ -124,11 +135,28 @@ protected int uploadFile(OwnCloudClient client) throws IOException {
                         ", chunk index " + chunkIndex + ", count " + chunkCount +
                         ", HTTP result status " + status);
 
-                if (!isSuccess(status))
+                if (isSuccess(status)){
+                  successfulChunks.add(String.valueOf(chunkIndex));
+                } else {
+                    SharedPreferences.Editor editor = sharedPref.edit();
+                    editor.putStringSet(chunkId, successfulChunks);
+                    editor.commit();
```

Comment:
Not sure anymore, but I thought apply() is the preferred way since it works asynchronous.